extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheMongoResourceProcessor.java

```diff
@@ -129,12 +133,28 @@ ReflectiveHierarchyBuildItem registerForReflection(CombinedIndexBuildItem index)
     }
 
     @BuildStep
-    void unremoveableClients(BuildProducer<UnremovableBeanBuildItem> unremovable) {
-        unremovable.produce(
-                new UnremovableBeanBuildItem(
-                        new UnremovableBeanBuildItem.BeanClassNamesExclusion(new HashSet<>(
-                                Arrays.asList(MongoClient.class.getName(), ReactiveMongoClient.class.getName())))));
+    void unremoveableClients(BuildProducer<MongoUnremovableClientsBuildItem> unremovable) {
+        unremovable.produce(new MongoUnremovableClientsBuildItem());
+    }
 
+    @BuildStep
+    public void mongoClientNames(ApplicationArchivesBuildItem applicationArchivesBuildItem,
```

Comment:
Shouldn't it be a `CombinedIndexBuildItem` to allow defining entities inside a third party library ?

Reply:
I pretty much did what we already do for `@MongoClientName`, so I propose we keep it like this for now and change in both places if it comes up in the future.

Reply:
OK, if (when) and issue arise we could fix this later